### PR TITLE
Build pure product, allowed mixed, native mixed and JMV mixed tests i…

### PR DESCRIPTION
**Pull request by ppalaga**
…n parallel

### CODE REVIEW

**Comment by ppalaga** [comment]
It looks like the state of the tree and local Maven repo is not propagated to the subsequent stages by default. Looks like we need to stash/unstash explicitly https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#stash-stash-some-files-to-be-used-later-in-the-build

**Comment by ppalaga** [comment]
@llowinge this is the idea. All tests are now linked from `product/*/group-*` modules. Tests are assigned to them in a round robin fashion with target group count 10 (configurable). Let's see how well it's going to work here.

File: Jenkinsfile.redhat [hunk 1]
```diff
@@ -45,66 +45,180 @@ pipeline {
...
-        stage('Build') {
-            steps {
-                configFileProvider([configFile(fileId: 'fuse-maven-settings', variable: 'MAVEN_SETTINGS')]) {
-                    sh "./mvnw ${MAVEN_PARAMS} clean install -Dquickly -Pmixed"
+        stage('Tests') {
+            parallel {
+                // %generated-stages-start%
+                stage('Product :: Group 01') {
+                    agent {
+                        label "rhel"
```
Comment (ppalaga) [inline Jenkinsfile.redhat:53]: Not sure whether `rhel` is the best choice here. It was the first label I saw in other Fuse pipelines.

**Comment by ppalaga** [comment]
Native mode is TBD

File: Jenkinsfile.redhat (hunk 1 above)
Comment (jamesnetherton) [reply Jenkinsfile.redhat:53]: TBH most of the labels are meaningless and result in execution on the same agent type. The reason we have so many is probably for legacy reasons.

**Pull request merged**